---
title: "Mo'orea & Tahiti 16s diversity analysis"
author: "Nicola G. Kriefall"
date: "7/21/2021"
output:
 rmarkdown::html_document:
    theme: cerulean
    toc: yes
    toc_float: yes
    highlight: haddock
    number_sections: true
---

```{r setup, include=FALSE}
#work computer 
setwd("/Volumes/GoogleDrive-104519233854090018057/My Drive/Moorea_revisions/moorea_holobiont_revised/mr16s_revised/02.diversity")
```

# Setup

## Packages

```{r packages}
library(ggplot2)
library(cowplot)
library(phyloseq)
library(car)
library(ggpubr)
library(vegan)
library(dada2)
```

## Read in data

```{r read in data}
samdf <- read.csv("mr16s_sampledata_plusneg copy.csv",header=TRUE)
row.names(samdf) <- samdf$id
load("taxa2 copy.Rdata")
```

## Phyloseq objects

```{r making phyloseq objects, eval=FALSE}
seqtab.rare <- read.csv("mr16s_seqtab.rev.cleanest.rare_6.2k copy.csv",row.names=1)
ps.rare <- phyloseq(otu_table(seqtab.rare, taxa_are_rows=FALSE), 
                    sample_data(samdf), 
                    tax_table(taxa2))
ps.rare 
#save(ps.rare,file="ps.rare.Rdata")
#1947 taxa during revisions - raw & rarefied

#raw, unrarefied version:
seqtab.clean <- read.csv("mr16s_seqtab.rev.cleanest copy.csv",row.names=1)
ps.clean <- phyloseq(otu_table(seqtab.clean, taxa_are_rows=FALSE), 
                    sample_data(samdf), 
                    tax_table(taxa2))
ps.clean
#save(ps.clean,file="ps.clean.Rdata")
#2346 taxa just raw, not rarefied

##TRIMMED VERSIONS:
seqtab.rare.trim <- read.csv("mr16s_seqtab.rev.trim.rare_6k copy.csv",row.names=1)
ps.rare.trim <- phyloseq(otu_table(seqtab.rare.trim, taxa_are_rows=FALSE), 
                    sample_data(samdf), 
                    tax_table(taxa2))
ps.rare.trim
#save(ps.rare.trim,file="ps.rare.trim.Rdata")
#223 taxa & 84 samples rarefied

#checking what happens if we read in the unrarefied version:
seqtab.trim <- read.csv("mr16s_seqtab.rev.cleanest.trim copy.csv",row.names=1)
ps.trim <- phyloseq(otu_table(seqtab.trim, taxa_are_rows=FALSE), 
                    sample_data(samdf), 
                    tax_table(taxa2))
ps.trim
#save(ps.trim,file="ps.trim.Rdata")
#223 taxa & 92 samples not rarefied
```

```{r read in ps objects}
load("ps.clean.Rdata")
load("ps.rare.Rdata")
load("ps.rare.trim.Rdata")
load("ps.trim.Rdata")
```

# Diversity

[Notes from phyloseq author](https://rdrr.io/bioc/phyloseq/man/estimate_richness.html)
Visualize alpha-diversity - Should be done on raw, untrimmed dataset

```{r generate div metrics}
df <- data.frame(estimate_richness(ps.rare, split=TRUE, measures=c("Shannon","InvSimpson","Observed")))
#df <- data.frame(estimate_richness(ps.clean, split=TRUE, measures=c("Shannon","InvSimpson","Observed")))
#df <- data.frame(estimate_richness(ps.trim, split=TRUE, measures=c("Shannon","InvSimpson","Observed")))
#df <- data.frame(estimate_richness(ps.rare.trim, split=TRUE, measures=c("Shannon","InvSimpson","Observed")))

df$id <- rownames(df)
df.div <- merge(df,samdf,by="id") #add sample data

#shannon diversity divided by species richness
df.div$even <- df.div$Shannon/(log(df.div$Observed))

df.div.mnw <- subset(df.div,site=="MNW")
df.div.mse <- subset(df.div,site=="MSE")
df.div.tnw <- subset(df.div,site=="TNW")
```

## Alpha metrics {.tabset}

Looking at everything individually, more of a synthesis below

### Shannon

```{r shannon site}
gg.site.sha <- ggplot(df.div,aes(x=site,y=Shannon,color=site))+
  geom_boxplot(outlier.shape=NA)+
  scale_color_manual(values=c("darkslategray3","darkslategray4","#000004"))+
  geom_jitter(alpha=0.5)+
  ylab("Shannon index")+
  xlab("Site")+
  theme_cowplot()+
  theme(legend.position="none")
gg.site.sha
```

```{r shannon rz}
#plotting reef zones
gg.sh <- ggplot(df.div, aes(x=zone, y=Shannon,color=zone,shape=zone))+
  geom_boxplot(outlier.shape=NA)+
  xlab("Reef zone")+
  ylab("Shannon index")+
  theme_cowplot()+
  scale_shape_manual(values=c(16,15),labels=c("BR","FR"))+
  scale_x_discrete(labels=c("BR","FR"))+
  scale_colour_manual(values=c("#ED7953FF","#8405A7FF"),labels=c("Back reef","Fore reef"))+
  #guides(color=guide_legend(title="Reef zone"),shape=guide_legend(title="Reef zone"))+
  theme(legend.position="none")+
  geom_jitter(alpha=0.5)+
  facet_wrap(~site)
gg.sh
```

```{r shannon stats}
#Shannon
shapiro.test(df.div$Shannon) #fine
leveneTest(df.div$Shannon~site/zone,data=df.div) #fine

a.div <- aov(Shannon~site/zone,data=df.div)
summary(a.div)
TukeyHSD(a.div) #ns for rarefied, trimmed, or not

#stats by site:
wilcox.test(Shannon~zone,data=df.div.mnw) #ns
summary(aov(Shannon~zone,data=df.div.mnw)) #ns
wilcox.test(Shannon~zone,data=df.div.mse) #ns
summary(aov(Shannon~zone,data=df.div.mse)) #ns
wilcox.test(Shannon~zone,data=df.div.tnw) #ns
summary(aov(Shannon~zone,data=df.div.tnw)) #ns
```

### Simpson

```{r simpson site}
gg.site.sim <- ggplot(df.div,aes(x=site,y=InvSimpson,color=site))+
  geom_boxplot(outlier.shape=NA)+
  scale_color_manual(values=c("darkslategray3","darkslategray4","#000004"))+
  geom_jitter(alpha=0.5)+
  ylab("Inv. Simpson index")+
  xlab("Site")+
  theme_cowplot()+
  theme(legend.position="none")
gg.site.sim
```

```{r simpson rz}
gg.si <- ggplot(df.div, aes(x=zone, y=InvSimpson,color=zone,shape=zone))+
  geom_boxplot(outlier.shape=NA)+
  xlab("Reef zone")+
  ylab("Inv. Simpson index")+
  theme_cowplot()+
  scale_shape_manual(values=c(16,15),labels=c("Back reef","Fore reef"))+
  scale_colour_manual(values=c("#ED7953FF","#8405A7FF"),labels=c("Back reef","Fore reef"))+
  #guides(color=guide_legend(title="Reef zone"),shape=guide_legend(title="Reef zone"))+
  theme(legend.position="none")+
  geom_jitter(alpha=0.5)+
  facet_wrap(~site)+
  scale_x_discrete(labels=c("BR","FR"))
gg.si
```

```{r simpson stats}
#simpson
shapiro.test(df.div$InvSimpson) #not normal
df.div$si.log <- log(df.div$InvSimpson) 
shapiro.test(df.div$si.log) #normal
leveneTest(df.div$si.log~site/zone,data=df.div) #fine 

a.div <- aov(si.log~site/zone,data=df.div)
summary(a.div)
TukeyHSD(a.div) #nothing, rarefied or not

#stats by site:
wilcox.test(InvSimpson~zone,data=df.div.mnw) #ns
summary(aov(InvSimpson~zone,data=df.div.mnw)) #ns
wilcox.test(InvSimpson~zone,data=df.div.mse) #ns
summary(aov(InvSimpson~zone,data=df.div.mse)) #ns
wilcox.test(InvSimpson~zone,data=df.div.tnw) #ns
summary(aov(log(InvSimpson)~zone,data=df.div.tnw)) #ns
```

### Richness

```{r richness site}
gg.site.obs <- ggplot(df.div,aes(x=site,y=Observed,color=site))+
  geom_boxplot(outlier.shape=NA)+
  scale_color_manual(values=c("darkslategray3","darkslategray4","#000004"))+
  geom_jitter(alpha=0.5)+
  ylab("ASV richness")+
  xlab("Site")+
  theme_cowplot()+
  theme(legend.position="none")
gg.site.obs
```

```{r richness rz}
gg.obs <- ggplot(df.div, aes(x=zone, y=Observed,color=zone,shape=zone))+
  geom_boxplot(outlier.shape=NA)+
  xlab("Reef zone")+
  ylab("ASV richness")+
  theme_cowplot()+
  scale_shape_manual(values=c(16,15),labels=c("Back reef","Fore reef"))+
  scale_colour_manual(values=c("#ED7953FF","#8405A7FF"),labels=c("Back reef","Fore reef"))+
  #guides(color=guide_legend(title="Reef zone"),shape=guide_legend(title="Reef zone"))+
  theme(legend.position="none")+
  geom_jitter(alpha=0.5)+
  facet_wrap(~site)+
  scale_x_discrete(labels=c("BR","FR"))
gg.obs
```

```{r richness stats}
#otu richness
shapiro.test(df.div$Observed) #nope
df.div$obs.log <- log(df.div$Observed)
shapiro.test(df.div$obs.log) #yessss
leveneTest(df.div$obs.log~site/zone,data=df.div) #tried site, zone, and both, was fine

a.div <- aov(obs.log~site/zone,data=df.div)
summary(a.div)
TukeyHSD(a.div) #Tahiti different from the other two 

#stats by site:
wilcox.test(Observed~zone,data=df.div.mnw) #ns
summary(aov(Observed~zone,data=df.div.mnw)) #ns
wilcox.test(Observed~zone,data=df.div.mse) #significant
# data:  Observed by zone
# W = 58, p-value = 0.04231
# alternative hypothesis: true location shift is not equal to 0
summary(aov(Observed~zone,data=df.div.mse)) #significant
#             Df Sum Sq Mean Sq F value Pr(>F)  
# zone         1   4041    4041   4.942 0.0348 *
# Residuals   27  22078     818                 
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
wilcox.test(Observed~zone,data=df.div.tnw) #ns
summary(aov(Observed~zone,data=df.div.tnw)) #ns
```

### Evenness

```{r evenness site}
gg.site.eve <- ggplot(df.div,aes(x=site,y=even,color=site))+
  geom_boxplot(outlier.shape=NA)+
  scale_color_manual(values=c("darkslategray3","darkslategray4","#000004"))+
  geom_jitter(alpha=0.5)+
  ylab("Evenness")+
  xlab("Site")+
  theme_cowplot()+
  theme(legend.position="none")
gg.site.eve
```

```{r evenness rz}
gg.ev <- ggplot(df.div, aes(x=zone, y=even,color=zone,shape=zone))+
  geom_boxplot(outlier.shape=NA)+
  xlab("Reef zone")+
  ylab("Evenness")+
  theme_cowplot()+
  scale_shape_manual(values=c(16,15),labels=c("Back reef","Fore reef"))+
  scale_colour_manual(values=c("#ED7953FF","#8405A7FF"),labels=c("Back reef","Fore reef"))+
  #guides(color=guide_legend(title="Reef zone"),shape=guide_legend(title="Reef zone"))+
  theme(legend.position="none")+
  geom_jitter(alpha=0.5)+
  facet_wrap(~site)+
  scale_x_discrete(labels=c("BR","FR"))
gg.ev
```

```{r evenness stats}
#evenness
shapiro.test(df.div$even) #great
leveneTest(df.div$even~site/zone,data=df.div) #tried site, zone, and both, was fine

a.div <- aov(even~site/zone,data=df.div)
summary(a.div)
TukeyHSD(a.div) #no differences, rarefied or not

wilcox.test(even~zone,data=df.div.mnw) #ns
summary(aov(even~zone,data=df.div.mnw)) #ns
wilcox.test(even~zone,data=df.div.mse) #ns
summary(aov(even~zone,data=df.div.mse)) #ns
wilcox.test(even~zone,data=df.div.tnw) #ns
summary(aov(even~zone,data=df.div.tnw)) #ns
```

## Phylogenetic diversity (Faith's D)

Tutorial from dada2 author [here](https://f1000research.com/articles/5-1492/v2)

```{r packages phylo d}
#install.packages('devtools')
#library(devtools)
#devtools::install_github('twbattaglia/btools')
library(btools)
```

(I'm not running the following chunk every time, because only need to generate the file once)

```{r fasta file generation, eval=FALSE}
# rare.otu <- as.matrix(ps.rare@otu_table)
# rare.taxa <- data.frame(ps.rare@tax_table)
# rownames(rare.taxa)==colnames(rare.otu)
# 
# colnames(rare.otu) <- rare.taxa$V8
# ids <- rownames(rare.taxa)
# 
# path="~/nicfall drive/Moorea_revisions/mr16s_revised/analyze_asv_table/mr16s_rev.cleanest.fasta"
# uniquesToFasta(rare.otu, path, ids = ids, mode = "w", width = 20000)
```

Actual analysis part: 

(I'm not running the following chunk every time, because only need to generate the files once)

```{r phylo d, eval=FALSE}
# seqs <- getSequences("mr16s_rev.cleanest.fasta")
# names(seqs) <- seqs # This propagates to the tip labels of the tree
# saveRDS(seqs,file="phylo.seqs.rev.cleanest.rds")

# #also doing the same thing with a .fasta file post-trimming to see if it makes a difference:
# seqs <- getSequences("mr16s_rev.cleanest.trimmed copy.fasta")
# names(seqs) <- seqs # This propagates to the tip labels of the tree
# saveRDS(seqs,file="phylo.seqs.rev.cleanest.trimmed.rds")
```

Doing this next part in the cluster because it takes forever

```{bash terminal phylo things, eval=FALSE}
##script phylo.R looks like this now:
# library(dada2)
# library(phangorn)
# library(DECIPHER)
#
# seqs <- readRDS("./phylo.seqs.rev.cleanest.rds")
# alignment <- AlignSeqs(DNAStringSet(seqs), anchor=NA)
# phang.align <- phyDat(as(alignment, "matrix"), type="DNA")
# dm <- dist.ml(phang.align)
# treeNJ <- NJ(dm) # Note, tip order != sequence order
# fit = pml(treeNJ, data=phang.align)
#
# ## negative edges length changed to 0!
# 
# fitGTR <- update(fit, k=4, inv=0.2)
# fitGTR <- optim.pml(fitGTR, model="GTR", optInv=TRUE, optGamma=TRUE,
#                     rearrangement = "stochastic", control = pml.control(trace = 0))
# detach("package:phangorn", unload=TRUE)
# saveRDS(fitGTR, file="./phylo.fitgtr.rev.cleanest.rds")

nano phylo.sh

##job file looks like this:
#$ -V # inherit the submission environment
#$ -cwd # start job in submission directory
#$ -N phylo.sh # job name, anything you want
#$ -l h_rt=24:00:00
#$ -M thenicolakriefall@gmail.com
#$ -m be

module load R
module load rstudio
Rscript phylo.R
##exit

##on cluster:
qsub phylo.sh
##saved output as: phylo.fitgtr.rev.cleanest.rds
```

Back in R

```{r back in R}
library(btools)

fitGTR <- readRDS("phylo.fitgtr.rev.cleanest.rds")

#to check for effects of rarefying:
#ps.rare <- ps.clean 

#new phyloseq object:
taxa.rare <- data.frame(ps.rare@tax_table)
seqtab.rare <- data.frame(ps.rare@otu_table)
taxa.rare$sqs <- row.names(taxa.rare) 
taxa.rare$sqs == colnames(seqtab.rare)
row.names(taxa.rare) <- taxa.rare$V8
colnames(seqtab.rare) <- taxa.rare$V8
row.names(taxa.rare) == colnames(seqtab.rare)
taxa.rare <- as.matrix(taxa.rare)

ps.rare.tree <- phyloseq(otu_table(seqtab.rare, taxa_are_rows = FALSE),
                         sample_data(samdf),
                         tax_table(taxa.rare),
                         phy_tree(fitGTR$tree))

pd.div <- estimate_pd(ps.rare.tree)
row.names(df.div) <- df.div$id
df.div.pd <- merge(df.div,pd.div,by=0)

## saving diversity data frame ##
#save & read back in as needed
#write.csv(df.div.pd,file="mr16s_diversity_rev.rare_6.2k.csv") #saving
#df.div <- read.csv("mr16s_diversity_rev.rare_6.2k.csv",row.names=1,header=TRUE) #reading back in

##post trimming:
# fitGTR.trim <- readRDS("phylo.fitgtr.rev.cleanest.trimmed.rds")
# 
# #new phyloseq object:
# taxa.rare <- data.frame(ps.rare.trim@tax_table)
# seqtab.rare <- data.frame(ps.rare.trim@otu_table)
# taxa.rare$sqs <- row.names(taxa.rare) 
# taxa.rare$sqs == colnames(seqtab.rare)
# row.names(taxa.rare) <- taxa.rare$V8
# colnames(seqtab.rare) <- taxa.rare$V8
# row.names(taxa.rare) == colnames(seqtab.rare)
# taxa.rare <- as.matrix(taxa.rare)
# 
# ps.rare.tree <- phyloseq(otu_table(seqtab.rare, taxa_are_rows = FALSE),
#                          sample_data(samdf),
#                          tax_table(taxa.rare),
#                          phy_tree(fitGTR.trim$tree))
# 
# pd.div <- estimate_pd(ps.rare.tree)
# row.names(df.div) <- df.div$id
# df.div.pd <- merge(df.div,pd.div,by=0)
```

### Faith's D Plots {.tabset}

#### Figure - site

```{r faiths d site}
gg.fd.site <- ggplot(df.div.pd,aes(x=site,y=PD,color=site))+
  geom_boxplot(outlier.shape=NA)+
  scale_color_manual(values=c("darkslategray3","darkslategray4","#000004"))+
  geom_jitter(alpha=0.5)+
  ylab("Faith's D")+
  xlab("Site")+
  theme_cowplot()+
  theme(legend.position="none")
gg.fd.site
```

#### Figure - reef zone

```{r faiths d rz}
gg.fd.rz <- ggplot(df.div.pd, aes(x=zone, y=PD,color=zone,shape=zone))+
  geom_boxplot(outlier.shape=NA)+
  xlab("Reef zone")+
  ylab("Faith's D")+
  theme_cowplot()+
  scale_shape_manual(values=c(16,15),labels=c("Back reef","Fore reef"))+
  scale_colour_manual(values=c("#ED7953FF","#8405A7FF"),labels=c("Back reef","Fore reef"))+
  #guides(color=guide_legend(title="Reef zone"),shape=guide_legend(title="Reef zone"))+
  theme(legend.position="none")+
  geom_jitter(alpha=0.5)+
  facet_wrap(~site)+
  scale_x_discrete(labels=c("BR","FR"))
gg.fd.rz
```

#### Stats

```{r phylo div stats}
#stats - phylo div
shapiro.test(log(df.div.pd$PD)) #normal
df.div.pd$faith.log <- log(df.div.pd$PD) 
leveneTest(faith.log~site*zone,data=df.div.pd) #fine 

a.div <- aov(faith.log~site/zone,data=df.div.pd)
summary(a.div)
TukeyHSD(a.div) #Tahiti lower than the other two

df.div.pd.mnw <- subset(df.div.pd,site=="MNW")
df.div.pd.mse <- subset(df.div.pd,site=="MSE")
df.div.pd.tnw <- subset(df.div.pd,site=="TNW")

wilcox.test(PD~zone,data=df.div.pd.mnw) #ns
summary(aov(PD~zone,data=df.div.pd.mnw)) #ns
wilcox.test(PD~zone,data=df.div.pd.mse) #ns
summary(aov(PD~zone,data=df.div.pd.mse)) #ns
wilcox.test(PD~zone,data=df.div.pd.tnw) #ns
summary(aov(PD~zone,data=df.div.pd.tnw)) #ns
```

## Synthesizing results {.tabset}

Only significant comparison above is Tahiti being different from Mo'orea NW & SE in ASV richness & phylo diversity (significantly lower)

### Plot by site

```{r plots site}
gg.panels.site <- ggarrange(gg.site.obs,gg.site.eve,gg.site.sha,gg.site.sim,gg.fd.site,nrow=3,ncol=2,labels="AUTO")
gg.panels.site
#ggsave(gg.panels.site,file="bac.div.site.pdf",height=8)
```

### Plot by reef zone

```{r plots rz}
gg.panels.rz <- ggarrange(gg.obs,gg.ev,gg.sh,gg.si,gg.fd.rz,nrow=3,ncol=2,labels="AUTO")
gg.panels.rz
#ggsave(gg.panels.rz,file="bac.div.rz.pdf", height=7)
```








